### PR TITLE
RDCC-5489 - Staff : configure domain as list

### PR DESCRIPTION
**Pull request by vilasshelke-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-5489

### Change description ###

Staff : configure domain as list

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/util/EmailValidator.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Override
+    public void initialize(ValidateEmail validateEmail) {
+        this.emailDomainList = EmailDomainPropertyInitiator.emailDomains;
+    }
+
+    @Override
+    public boolean isValid(String emailAddress, ConstraintValidatorContext constraintValidatorContext) {
+        if (ObjectUtils.isEmpty(emailAddress) || ObjectUtils.isEmpty(this.emailDomainList)) {
+            log.error("Email id => '{}' or email.domainList => {} should not empty", emailAddress, emailDomainList);
```
Comment (SujeethKumarAri) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/util/EmailValidator.java:30]: Can you please remove logging the email address as it is PII data?

**Review by SujeethKumarAri — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/util/EmailValidator.java (hunk 1 above)
Comment (vilasshelke-hmcts) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/util/EmailValidator.java]: Ok

**Review by SujeethKumarAri — APPROVED** [review]

**Review by sahitya-desireddy — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/cwrdapi/util/EmailValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    @Test
+    void testInValid_EmailNotSet_Expected_False() {
+        //if email is empty or null it should be false
+        assertThat(emailValidator.isValid(null, context)).isFalse();
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"@justice.gov.uk", "vilas.shelke 123@justice.gov.uk", "vilas.shelke@gmail.com"})
+    void testInValid_EmailWithoutFirstName_Expected_False() {
```
Comment (rajkatla-hmcts) [inline src/test/java/uk/gov/hmcts/reform/cwrdapi/util/EmailValidatorTest.java:52]: parameters not used in test case

Comment (vilasshelke-hmcts) [reply src/test/java/uk/gov/hmcts/reform/cwrdapi/util/EmailValidatorTest.java:52]: @rajkatla-hmcts - this was sonar violations. As per sonar practice it fixed. In ValueSource parameters are applied.

Comment (rajkatla-hmcts) [reply src/test/java/uk/gov/hmcts/reform/cwrdapi/util/EmailValidatorTest.java:52]: @vilasshelke-hmcts which one sonar violation ? if you are not using value from source then what is the point of making it as parameterized test ?

Comment (vilasshelke-hmcts) [reply src/test/java/uk/gov/hmcts/reform/cwrdapi/util/EmailValidatorTest.java:52]: @rajkatla-hmcts -  @ValueSource(strings = {"@justice.gov.uk", "vilas.shelke 123@justice.gov.uk", "vilas.shelke@gmail.com"}) these are the values

Comment (vilasshelke-hmcts) [reply src/test/java/uk/gov/hmcts/reform/cwrdapi/util/EmailValidatorTest.java]: @rajkatla-hmcts - modified test case

**Review by rajkatla-hmcts — APPROVED** [review]

**Review by sahitya-desireddy — APPROVED** [review]

**Pull request merged**